components/tidb_query/src/expr/builtin_string.rs

```diff
@@ -492,17 +483,13 @@ impl ScalarFunc {
         if self.field_type.get_flen() == -1
             || self.field_type.get_flen() > tidb_query_datatype::MAX_BLOB_WIDTH
         {
-            return Ok(Some(Cow::Borrowed(b"")));
+            return Ok(Some(Cow::Owned(b"".to_vec())));
```

Comment:
Actually this is a bug( I implemented before 😭 ). It should return None here as the vectorize version you’d implemented. line 492 below should return `None` too. So you can move more logic to `expr_utils::string::to_base64`.
And finally only code like `Ok(expr_utils::string::to_abas64(s.as_ref(), s.field_type.get_flen()).map(Cow::Owned)` left here.